src/pybind/mgr/volumes/fs/health.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import threading
+
+from .exception import VolumeException
+
+log = logging.getLogger(__name__)
+
+# singleton design pattern taken from http://www.aleax.it/5ep.html
```

Comment:
Nice ! I found this write up easier to follow, https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html